### PR TITLE
Allow region to be specified via env vars

### PR DESCRIPTION
**Pull request by paulgb**
- Allow region to be specified via env vars.
- Remove some panics and add some error context.
- Fixed an issue where 404s were not being propagated correctly from s3-rust because a feature flag was disabled.

### CODE REVIEW

File: crates/y-sweet-core/src/sync_kv.rs [hunk 1]
```diff
@@ -26,12 +26,13 @@ impl SyncKv {
...
-        let data = if let Some(snapshot) = store.get(&key).await? {
-            tracing::info!(size=?snapshot.len(), "Loaded snapshot");
-            bincode::deserialize(&snapshot)?
-        } else {
-            BTreeMap::new()
-        };
+        let data =
+            if let Some(snapshot) = store.get(&key).await.context("Failed to get from store.")? {
+                tracing::info!(size=?snapshot.len(), "Loaded snapshot");
+                bincode::deserialize(&snapshot).context("Failed to deserialize.")?
```
Comment (rolyatmax) [inline crates/y-sweet-core/src/sync_kv.rs:32]: Might be worth logging more context here and above so we can look into which key didn't exist or which key's value failed to deserialize?

File: crates/y-sweet/src/main.rs [hunk 2]
```diff
@@ -51,12 +51,18 @@ enum ServSubcommand {
...
     if store_path.starts_with("s3://") {
-        let region = Region::UsEast1;
-
+        let region = match Region::from_default_env() {
+            Ok(region) => {
+                tracing::info!(region=?region, "Using region from environment.");
+                region
+            }
+            Err(e) => {
+                tracing::warn!(error=?e, "Failed to get region from environment, using default.");
```
Comment (rolyatmax) [inline crates/y-sweet/src/main.rs:60]: Worth mentioning here that the default value is us-east-1?

**Review by rolyatmax — APPROVED** [review]

File: crates/y-sweet/src/main.rs (hunk 2 above)
Comment (paulgb) [reply crates/y-sweet/src/main.rs]: Done.

File: crates/y-sweet-core/src/sync_kv.rs (hunk 1 above)
Comment (rolyatmax) [reply crates/y-sweet-core/src/sync_kv.rs:32]: Gotcha - does that include the key or the value?

**Pull request merged**